src/Groups/matrices/forms.jl

```diff
@@ -467,3 +468,399 @@ function issingular(f::SesquilinearForm{T}) where T
+      MM = zero_matrix(F,n^2,n^2)
+      for i in 1:n, j in 1:n, k in 1:n
+         MM[n*(i-1)+j,n*(k-1)+j] += mat[i,k]
+         MM[n*(i-1)+j,n*(i-1)+k] -= mmat[j,k]
+      end
+      push!(M,MM)
+   end
+
+   r,K = nullspace(block_matrix(length(M),1,M))
+   
+   return [matrix(F,n,n,[K[i,j] for i in 1:n^2]) for j in 1:r]
+end
+
+"""
+    invariant_quadratic_forms(G::MatrixGroup)
+
+Return a generating set for the vector spaces of quadratic forms preserved by the group `G`.
+!!! warning "Note:"
+    At the moment, elements of the generating set are returned of type `mat_elem_type(G)`.
+"""
+function invariant_quadratic_forms(G::MatrixGroup{S,T}) where {S,T}
+   F = base_ring(G)
+   n = degree(G)
+   M = T[]
+   for mat in gens(G)
+      MM = zero_matrix(F,div(n*(n+1),2),div(n*(n+1),2))
+      for i in 1:n
+      for p in 1:n, q in p:n
+         MM[div((2*n-i+2)*(i-1),2)+1, div((2*n-p+2)*(p-1),2)+q-p+1] = mat[i,p]*mat[i,q]
+      end
+      for j in i+1:n, p in 1:n, q in p:n
+         MM[div((2*n-i+2)*(i-1),2)+j-i+1, div((2*n-p+2)*(p-1),2)+q-p+1] = mat[i,p]*mat[j,q]+mat[j,p]*mat[i,q]
+      end
+      end
+      for i in 1:div(n*(n+1),2) MM[i,i]-=1 end
+      push!(M,MM)
+   end
+
+   r,K = nullspace(block_matrix(length(M),1,M))
+   M = T[]
+   for i in 1:r
+      push!(M,upper_triangular_matrix([K[j,i] for j in 1:div(n*(n+1),2)]))
+   end
+   return M
+end
+
+#TODO: do we want to keep these?
+#TODO: hermitian forms require resolution of semilinear systems
+"""
+    invariant_symmetric_forms(G::MatrixGroup)
+
+Return a generating set for the vector spaces of symmetric forms preserved by the group `G`.
+!!! warning "Note:"
+    At the moment, elements of the generating set are returned of type `mat_elem_type(G)`.
+"""
+function invariant_symmetric_forms(G::MatrixGroup{S,T}) where {S,T}
+   F = base_ring(G)
+   n = degree(G)
+   M = T[]
+
+   for mat in gens(G)
+      MM = zero_matrix(F,div(n*(n+1),2),div(n*(n+1),2))
+      idx_r=1
+      for i in 1:n, j in i:n
+         idx_c=1
+         for s in 1:n
+            MM[idx_r,idx_c] = mat[i,s]*mat[j,s]
+            idx_c+=1
+            for t in (s+1):n
+               MM[idx_r,idx_c] = mat[i,s]*mat[j,t]+mat[i,t]*mat[j,s]
+               idx_c+=1
+            end
+         end
+         idx_r+=1
+      end
+      MM -= one(MM)
+      push!(M,MM)
+   end
+
+   r,K = nullspace(block_matrix(length(M),1,M))
+
+   L = T[]
+   for j in 1:r
+      B = upper_triangular_matrix([K[i,j] for i in 1:div(n*(n+1),2)])
+      for i in 1:n, j in 1:i-1
+         B[i,j]=B[j,i]
+      end
+      push!(L, B)
+   end
+   return L
+end
+
+"""
+    invariant_alternating_forms(G::MatrixGroup)
+
+Return a generating set for the vector spaces of alternating forms preserved by the group `G`.
+!!! warning "Note:"
+    At the moment, elements of the generating set are returned of type `mat_elem_type(G)`.
+"""
+function invariant_alternating_forms(G::MatrixGroup{S,T}) where {S,T}
```

Comment:
```suggestion
function invariant_alternating_forms(G::MatrixGroup)
```

Reply:
I used `T` in the third line of the function, where I define `M = T[]`. Do I remove the parameters and replace the third line by `M = mat_elem_type(G)` ?
(the same correction will apply to the other analogous functions)